presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -285,6 +286,53 @@ public PlanNode visitExchange(ExchangeNode node, RewriteContext<Void> context)
             return new ExchangeNode(node.getId(), node.getType(), node.getScope(), partitioningScheme, sources, inputs);
         }
 
+        private void mapExchangeNodeSymbols(ExchangeNode node)
+        {
+            if (node.getInputs().size() == 1) {
+                for (int symbolIndex = 0; symbolIndex < node.getOutputSymbols().size(); symbolIndex++) {
+                    Symbol canonicalOutput = canonicalize(node.getOutputSymbols().get(symbolIndex));
+                    Symbol canonicalInput = canonicalize(node.getInputs().get(0).get(symbolIndex));
+
+                    if (!canonicalOutput.equals(canonicalInput)) {
+                        map(canonicalOutput, canonicalInput);
+                    }
+                }
+                return;
+            }
+
+            // Mapping from list [node.getInput(0).get(symbolIndex), node.getInput(1).get(symbolIndex), ...] to node.getOutputSymbols(symbolIndex).
+            // All symbols are canonical.
+            Map<List<Symbol>, Symbol> inputsToOutputs = new HashMap<>();
+            for (int symbolIndex = 0; symbolIndex < node.getOutputSymbols().size(); symbolIndex++) {
```

Comment:
Added comment: `Map each same list of input symbols [I1, I2, ..., In] to the same output symbol O`